openrtb-core/src/main/protobuf/openrtb.proto

```diff
@@ -995,6 +999,76 @@ message BidRequest {
+    //
+    // Note: When available, fields are sourced from Client Hints HTTP headers
+    // or equivalent JavaScript accessors from the NavigatorUAData interface.
+    // For agents that have no support for User-Agent Client Hints, an exchange
+    // can also extract information from the parsed User-Agent header, so this
+    // object can always be used as the source of the user agent information.
+    message UserAgent {
+      // Identifies a device's browser or similar software component, and the
+      // user agent's execution platform or operating system.
+      message BrandVersion {
+        // A brand identifier, for example, "Chrome" or "Windows". The value may
+        // be sourced from the User-Agent Client Hints headers, representing
+        // either the user agent brand (from the Sec-CH-UA-Full-Version header)
+        // or the platform brand (from the Sec-CH-UA-Platform header).
+        // Not supported by Google.
+        optional string brand = 1;
+
+        // A sequence of version components, in descending hierarchical order
+        // (major, minor, micro, ...).
+        // Not supported by Google.
+        repeated string version = 2;
+      }
+      // Each BrandVersion object identifies a browser or similar software
+      // component. Exchanges should send brands and versions derived from
+      // the Sec-CH-UA-Full-Version-List header.
+      // Not supported by Google.
+      repeated BrandVersion browsers = 1;
+
+      // Identifies the user agent's execution platform / OS. Exchanges should
+      // send a brand derived from the Sec-CH-UA-Platform header, and version
+      // derived from the Sec-CH-UAPlatform-Version header.
+      // Not supported by Google.
+      optional BrandVersion platform = 2;
+
+      // true if the agent prefers a "mobile" version of the content if
+      // available, meaning optimized for small screens or touch input. false if
+      // the agent prefers the "desktop" or "full" content. Exchanges should
+      // derive this value from the Sec-CH-UAMobile header.
+      // Not supported by Google.
+      optional bool mobile = 3;
+
+      // Device's major binary architecture, for example, "x86" or "arm".
+      // Exchanges should retrieve this value from the Sec-CH-UA-Arch header.
+      // Not supported by Google.
+      optional string architecture = 4;
+
+      // Device's bitness, for example, "64" for 64-bit architecture. Exchanges
+      // should retrieve this value from the Sec-CH-UA-Bitness header.
+      // Not supported by Google.
+      optional string bitness = 5;
+
+      // Device model. Exchanges should retrieve this value from the
+      // Sec-CH-UAModel header.
+      // Not supported by Google.
+      optional string model = 6;
+
+      // The source of data for the User Agent information.
+      // Not supported by Google.
+      optional UserAgentSource source = 7;
+    }
```

Comment:
would it make sense to open the door for extensions in case an exchange requires them in the future?

Reply:
There is already an extension field at Device level. So far OpenRTB does not support extensions here.

Reply:
true, I had to do some fine tuning to allow it for vungle, nvm then